src/python/pants/backend/jvm/tasks/junit_run.py

```diff
@@ -644,23 +661,46 @@ def _isolation(self, all_targets):
     try:
       yield output_dir, reports, coverage
     finally:
-      # NB: Deposit of the "current" test output in the root workdir (.pants.d/test/junit) is a
-      # defacto public API and so we implement that behavior here to maintain backwards
-      # compatibility for non-pants report file consumers.
-      # TODO(John Sirois): Deprecate this ~API and provide a stable directory solution for test
-      # output: https://github.com/pantsbuild/pants/issues/3879
       lock_file = '.file_lock'
       with OwnerPrintingInterProcessFileLock(os.path.join(self.workdir, lock_file)):
-        # Kill everything except the isolated `_runs/` dir.
-        for name in os.listdir(self.workdir):
-          path = os.path.join(self.workdir, name)
-          if name not in (run_dir, lock_file):
-            if os.path.isdir(path):
-              safe_rmtree(path)
-            else:
-              os.unlink(path)
+        preserve = (run_dir, lock_file)
+        self._link_current_reports(output_dir, preserve)
+
+  def _link_current_reports(self, output_dir, preserve):
+    # NB: Deposit of the "current" test output in the root workdir (.pants.d/test/junit) is a
+    # defacto public API and so we implement that behavior here to maintain backwards
+    # compatibility for non-pants report file consumers.
+    # TODO(John Sirois): Deprecate this ~API and provide a stable directory solution for test
+    # output: https://github.com/pantsbuild/pants/issues/3879
+
+    # Kill everything except the isolated `_runs/` dir.
```

Comment:
nit: maybe update to `kill everything except the preserved dirs`.

Reply:
Fixed.